numba/experimental/jitclass/boxing.py

```diff
@@ -144,12 +144,14 @@ def _specialize_box(typ):
         "__ixor__",
     }
     for name, func in typ.methods.items():
+        if name == "__init__":
+            continue
         if (
             name.startswith("__")
             and name.endswith("__")
             and name not in supported_dunders
         ):
-            continue
+            raise TypeError("Method {0} is not supported".format(name))
```

Comment:
Suggest adding a unit test that checks this error message is raised as expected to `numba.tests.test_jitclasses`.